### ISSUE TITLE
Idea:  JarJar guava?

### ISSUE THREAD

**Issue by terabyte** [RisingOak/jenkins-client#75]
Hey Cosmin,

For my usecase, I require an ancient, stupid version of guava.  I've managed to make things work for me using the jarjar plugin to bundle jenkins-client's dependency on guava.  Is this something I should just continue to do myself, or do you think it makes sense to do this in the main project?

I would completely understand it if this dependency tragedy is something you think should stay outside the library - but if other people also have this issue, it may help to bundle up our dependencies.

Thoughts?

**Comment by khmarbaise** [comment]
I would say you can simply do an exclusion in your pom...If you build works so i don't see a problem....May be i misunderstand a thing here? Can you make a suggestion what you thing how it looks like?

**Comment by terabyte** [comment]
so here is the situation:  I am writing a plugin for atlassian stash.  Stash depends upon guava-10.0.something.  It doesn't work with newer guava.  This library depends upon guava-not-old-as-shit.  It doesn't work with old-ass-guava.

So if I exclude *either* version, and use the other, something breaks.  I need this library to get its guava via jar-jar, without "polluting" the classpath with new guava.

I have a proof of concept working here:
https://github.com/terabyte/stashbot/commit/d8b2c88a88a1fef4992ddd0c2803e30803e02c19

See if you can make that project build and work without using a jar-jar'd guava for jenkins-client.jar.  I couldn't.

The code changes necessary to build the custom jenkins-client.jar are in this change, it was super simple:
https://github.com/terabyte/jenkins-client/commit/4f8ce857b5abf662129e8f95f1815db928cc64ea

We would probably want it to only jar-jar guava if the user asked for it. so making that conditional might be *slightly* more involved?

**Comment by cosmin** [comment]
That is actually what I originally wrote this library for, a Stash plugin to check the build status in Jenkins. I used this to prevent people from pushing if the build was broken, unless they claim to be fixing the broken build.

So I can relate to this need. I'm guessing maybe I started with an older Guava back then.

I think using jarjar to package up guava under a different root is probably a good idea.
